pkg/secrethub/credentials/encoding.go

```diff
@@ -43,7 +43,7 @@ type EncodableCredential interface {
 	// Export exports the credential in a format that can be decoded by its Decoder.
 	Export() []byte
 	// Decoder returns a decoder that can decode an exported key back into a Credential.
-	Decoder() CredentialDecoder
+	Decoder() credentialDecoder
```

Comment:
Not sure, but I think the linter won't like this. Returning a private type from a public function is generally not accepted.